Common/src/interface_interpolation/CInterpolator.cpp

```diff
@@ -143,253 +130,253 @@ unsigned long CInterpolator::Collect_ElementInfo(int markDonor, unsigned short n
-    if( Aux_Send_Map[iVertex] != nullptr )
-      delete [] Aux_Send_Map[iVertex];
+  // global point IDs of surface-neighbors of all domain vertices on the marker
+  su2vector<unsigned long> Buffer_Send_LinkedNodes(nLocalLinkedNodes);
+  unsigned long index = 0;
+  for (iVertex = 0; iVertex < nVertex; iVertex++) {
+    iPoint = geom->vertex[val_marker][iVertex]->GetNode();
+    if (geom->nodes->GetDomain(iPoint)) {
+      for (unsigned long jPoint_global : neighbors[iPoint]) {
+        Buffer_Send_LinkedNodes[index] = jPoint_global;
+        index++;
+      }
+    }
   }
-  delete [] Aux_Send_Map; Aux_Send_Map = nullptr;
 
-  /*--- Reconstruct  boundary by gathering data from all ranks ---*/
+  /*--- Now these arrays of all processes must be joined to a single/global arrays. For this,
+   * the entries of StartLinkedNodes must be shifted.
+   * Furthermore, the global point IDs in LinkedNodes are replaced by global vertex IDs.
+   * For this, the master process collects the data from all processes, joins them and broadcasts them again. ---*/
 
-  SU2_MPI::Allreduce(     &nLocalVertex,      &nGlobalVertex, 1, MPI_UNSIGNED_LONG, MPI_SUM, SU2_MPI::GetComm());
+  /*--- Allocate global arrays. ---*/
+  SU2_MPI::Allreduce(&nLocalVertex, &nGlobalVertex, 1, MPI_UNSIGNED_LONG, MPI_SUM, SU2_MPI::GetComm());
   SU2_MPI::Allreduce(&nLocalLinkedNodes, &nGlobalLinkedNodes, 1, MPI_UNSIGNED_LONG, MPI_SUM, SU2_MPI::GetComm());
 
-  Buffer_Receive_Coord       = new su2double    [ nGlobalVertex * nDim ];
-  Buffer_Receive_GlobalPoint = new long[ nGlobalVertex ];
-  Buffer_Receive_Proc        = new unsigned long[ nGlobalVertex ];
+  Buffer_Receive_Coord.resize(nGlobalVertex, nDim);
+  Buffer_Receive_GlobalPoint.resize(nGlobalVertex);
+  Buffer_Receive_Proc.resize(nGlobalVertex);
 
-  Buffer_Receive_nLinkedNodes     = new unsigned long[ nGlobalVertex ];
-  Buffer_Receive_LinkedNodes      = new unsigned long[ nGlobalLinkedNodes   ];
-  Buffer_Receive_StartLinkedNodes = new unsigned long[ nGlobalVertex ];
+  Buffer_Receive_nLinkedNodes.resize(nGlobalVertex);
+  Buffer_Receive_LinkedNodes.resize(nGlobalLinkedNodes);
+  Buffer_Receive_StartLinkedNodes.resize(nGlobalVertex);
 
+  /*--- Master process gathers data from all ranks ---*/
 #ifdef HAVE_MPI
-  if (rank == MASTER_NODE){
-
-    for (iVertex = 0; iVertex < nDim*nLocalVertex; iVertex++)
-      Buffer_Receive_Coord[iVertex]  = Buffer_Send_Coord[iVertex];
-
-    for (iVertex = 0; iVertex < nLocalVertex; iVertex++){
-      Buffer_Receive_GlobalPoint[iVertex]      = Buffer_Send_GlobalPoint[iVertex];
-      Buffer_Receive_Proc[iVertex]             = MASTER_NODE;
-      Buffer_Receive_nLinkedNodes[iVertex]     = Buffer_Send_nLinkedNodes[iVertex];
+  int nProcessor = size, iRank;
+  unsigned long received_nLocalLinkedNodes, received_nLocalVertex;
+
+  if (rank == MASTER_NODE) {
+    /*--- "Receive" from master process, i.e. copy. ---*/
+    for (unsigned long iVertex = 0; iVertex < nLocalVertex; iVertex++)
```

Comment:
To copy you can assign the matrix type directly which is more efficient than looping over each dimension

Reply:
I want to copy the data of `Buffer_Send_Coord` into the beginning of `Buffer_Receive_Coord`. `operator=` would resize (shrink) the receive buffer container and resizing (making larger) afterwards might invalidate the content?

Reply:
Fair enough, I thought they were the same size all the time.